From 4ae49076d17491341b02e9cb8a4fd28254e0616f Mon Sep 17 00:00:00 2001
From: "Dr. Git" <drgitx@gmail.com>
Date: Thu, 20 May 2021 15:35:05 +0000
Subject: [PATCH 058/143] ASoC: qcom: sdm845: set quaternary MI2S codec DAI to
 I2S format

Required for MAX98927 speaker audio to work on the OnePlus 6.
---
 sound/soc/qcom/sdm845.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sound/soc/qcom/sdm845.c b/sound/soc/qcom/sdm845.c
index a479d7e5b7..5cd6d34345 100644
--- a/sound/soc/qcom/sdm845.c
+++ b/sound/soc/qcom/sdm845.c
@@ -367,10 +367,12 @@ static int sdm845_snd_startup(struct snd_pcm_substream *substream)
 		snd_soc_dai_set_fmt(codec_dai, codec_dai_fmt);
 		break;
 	case QUATERNARY_MI2S_RX:
+		codec_dai_fmt |= SND_SOC_DAIFMT_NB_NF | SND_SOC_DAIFMT_I2S;
 		snd_soc_dai_set_sysclk(cpu_dai,
 			Q6AFE_LPASS_CLK_ID_QUAD_MI2S_IBIT,
 			MI2S_BCLK_RATE, SNDRV_PCM_STREAM_PLAYBACK);
 		snd_soc_dai_set_fmt(cpu_dai, fmt);
+		snd_soc_dai_set_fmt(codec_dai, codec_dai_fmt);
 
 
 		break;
-- 
2.49.0

